### PR TITLE
build(conan): fix conan cache invalidation caused by different build type detection

### PR DESCRIPTION
**Pull request by kstdl**
## Purpose

Dependencies from the conan was rebuilt on every build because of different build type detection by conan-cmake. Fixed by setting build type to the conan profile

### CODE REVIEW

**Review by agrebin — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -416,6 +416,16 @@ jobs:
...
       - checkout
+      - run:
+         name: Login into gcr
+         command: |
+            echo ${GCLOUD_SERVICE_KEY} | docker login -u _json_key --password-stdin https://gcr.io
+
+      - run:
+         name: Build docker builder image
+         command: |
+                docker build -t test --target builder .
```
Comment (sfrique) [inline .circleci/config.yml:427]: Are we sure we want this to run before the clean? 
To be honest, this cache should be automatically used, I don't see why we need to explicitly build this target.

Also, this uses a bad tag `test`

Comment (kstdl) [reply .circleci/config.yml:427]: As I saw docker removes build intermediate containers. So I thought builder image was just removed during full build. 
I will move it a little bit down after clean and will fix the tag. I changed it just for local test but forgot to change it back

File: .circleci/config.yml [hunk 2]
```diff
@@ -427,6 +427,17 @@ jobs:
...
+
+      - run:
+         name: Login into gcr
+         command: |
+            echo ${GCLOUD_SERVICE_KEY} | docker login -u _json_key --password-stdin https://gcr.io
+
+      - run:
+         name: Build docker builder image
+         command: |
+                docker build -t ${IMAGE}:${DOCKER_BRANCH_TAG}-${CIRCLE_BUILD_NUM} --target builder .
```
Comment (sfrique) [inline .circleci/config.yml:439]: Have to check if it works, but the `DOCKER_BRANCH_TAG`  is only generated a step later, right?

Comment (kstdl) [reply .circleci/config.yml:439]: Missed that. Thought it is a default circleci variable 
And it isnt working, because build failed

**Review by JakubFornadel — APPROVED** [review]

**Pull request merged**